src/main.rs

```diff
@@ -224,11 +224,13 @@ fn show_paths() {
     );
     let pages_dir = Cache::get_cache_dir()
         .map(|(path, _)| path.join("tldr-master"))
-        .map(|mut path| {
-            path.push(""); // Trailing path separator
-            path.to_str().unwrap_or("[Invalid]").to_string()
-        })
-        .unwrap_or_else(|e| format!("[Error: {}]", e));
+        .map_or_else(
+            |e| format!("[Error: {}]", e),
+            |mut path| {
+                path.push(""); // Trailing path separator
+                path.to_str().unwrap_or("[Invalid]").to_string()
```

Comment:
I think this should be changed further. The two calls to `map` are a bit weird and there are also two allocations that could be removed (`push` instead of `join` does not throw away the old buffer and converting the `PathBuf` to a `String` by using the inner `OsString`):

```rust
let pages_dir = Cache::get_cache_dir().map_or_else(
    |e| format!("[Error: {}]", e),
    |(mut path, _)| {
        path.push("tldr-master");
        path.push(""); // Trailing path separator
        path.into_os_string()
            .into_string()
            .unwrap_or_else(|_| String::from("[Invalid]"))
    },
);
```

Reply:
Similar changes can be made to the calls above, but I don't think they are mandatory.

Reply:
Applied. Thank you for your detailed description.